modules/cache/logic/cacheLogic.js

```diff
@@ -113,17 +114,13 @@ const CacheLogic = {
   async wordPriceHistory(wordSale) {
     const height = await aeternity.getHeight();
     const txs = await MdwLogic.getContractTransactions(height, 0, wordSale);
-    const eventsSchema = [
-      { name: 'Buy', types: [SOPHIA_TYPES.address, SOPHIA_TYPES.int, SOPHIA_TYPES.int] },
-      { name: 'Sell', types: [SOPHIA_TYPES.address, SOPHIA_TYPES.int, SOPHIA_TYPES.int] },
-    ];
 
     return txs.flatMap(tx => {
       const decodedEvent = () => {
-        const decodedEvents = decodeEvents(tx.tx.log, { schema: eventsSchema })
-          .flatMap(({ name, decoded }) => ({ name, decoded }))
-          .filter(log => log.decoded.length);
+        // FIXME: use ACI here
+        const decodedEvents = aeternity.decodeEvents(tx.tx.log, WORD_SALE_INTERFACE, 'TokenSale');
```

Comment:
tx return value now includes `.decodedEvents`

Reply:
Can you elaborate?